ShopperInsights/src/main/java/com/braintreepayments/api/ShopperInsightsClient.kt

```diff
@@ -103,16 +80,17 @@ class ShopperInsightsClient @VisibleForTesting internal constructor(
         callback: ShopperInsightsCallback
     ) {
         when {
-            error != null -> callback.onResult(ShopperInsightsResult.Failure(error))
-            result?.eligibleMethods?.paypal == null &&
-                    result?.eligibleMethods?.venmo == null -> {
-                callback.onResult(
-                    ShopperInsightsResult.Failure(
-                        BraintreeException("Required fields missing from API response body")
-                    )
+            error != null -> callbackFailure(callback, error)
+
+            result?.eligibleMethods?.paypal == null && result?.eligibleMethods?.venmo == null -> {
+                callbackFailure(
+                    callback = callback,
+                    error = BraintreeException("Required fields missing from API response body")
                 )
             }
+
             else -> {
+                braintreeClient.sendAnalyticsEvent(GET_RECOMMENDED_PAYMENTS_SUCCEEDED)
```

Comment:
Nit - I know we talked about this is another PR, but maybe we should align with the other payment method features also move the result return & analytics firing for the success case into a helper method similar to how we're doing it for the error case

Reply:
I had left out the success function since this is the only spot where we send the success event and invoke the success on the callback, but it looks like the other modules still have the success function. I'll update this to align!